Still can't compile on CentOS 7

Following @easyaspi314's commits in #465 which @easyaspi314 merged in 4aa3d59, the #464 situation is incrementally better now, in the respect that it looks like I can now manually turn off variants.

But out of the box, the dev branch (incl. later fixes) still won't compile on CentOS 7.

Different reason now, the intrinsics type isn't present but `XXH_DISPATCH_AVX2` is still on by default; `__AVX2__` isn't, so `immintrin.h` doesn't get included.

```
In file included from /tmp/src/xxHash/xxh_x86dispatch.c:208:0:
/tmp/src/xxHash/xxhash.h: In function 'XXH3_accumulate_512_avx2':
/tmp/src/xxHash/xxhash.h:3801:5: error: unknown type name '__m256i'
     {   XXH_ALIGN(32) __m256i* const xacc    =       (__m256i *) acc;
     ^
/tmp/src/xxHash/xxhash.h:3801:55: error: '__m256i' undeclared (first use in this function)
     {   XXH_ALIGN(32) __m256i* const xacc    =       (__m256i *) acc;
```

`immintrin.h` is in fact present on CentOS 7, so it's interesting that `__AVX2__` is not defined; `__SSE2__` is (and `emmintrin.h` does exist and get included).

Now, looking at the current logic:

```
#  if (defined(__GNUC__) \
       && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 7))) /* GCC 4.7+ */ \
   || (defined(_MSC_VER) && _MSC_VER >= 1900) /* VS 2015+ */ \
   || (defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 180030501) /* VS 2013 Update 2 */ \
   || XXH_HAS_INCLUDE(<avx2intrin.h>) /* GCC/Clang internal header */
```

I notice that `avx2intrin.h` does exist. But that's sort of irrelevant, because we're not including it, because there's completely separate logic:

```
#  if defined(__AVX2__)
#    include <immintrin.h>
#  elif defined(__SSE2__)
#    include <emmintrin.h>
```

IMHO we sort of need to go one way or the other, these two blocks need to effectively align.

--- a/xxh_x86dispatch.c
+++ b/xxh_x86dispatch.c
@@ -118,7 +118,8 @@ extern "C" {
  * @brief Enables/disables dispatching for AVX2.
  *
  * This is automatically detected if it is not defined.
- *  - GCC 4.7 and later are known to support AVX2.
+ *  - GCC 4.7 and later are known to support AVX2, but >4.9 is required for
+ *    to get the AVX2 intrinsics and typedefs without -mavx -mavx2.
  *  - Visual Studio 2013 Update 2 and later are known to support AVX2.
  *  - The GCC/Clang internal header `<avx2intrin.h>` is detected. While this is
  *    not allowed to be included directly, it still appears in the builtin
@@ -127,8 +128,7 @@ extern "C" {
  * @see XXH_AVX2
  */
 #ifndef XXH_DISPATCH_AVX2
-#  if (defined(__GNUC__) \
-       && (__GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 7))) /* GCC 4.7+ */ \
+#  if (defined(__GNUC__) && (__GNUC__ > 4)) /* GCC 5.0+ */ \
    || (defined(_MSC_VER) && _MSC_VER >= 1900) /* VS 2015+ */ \
    || (defined(_MSC_FULL_VER) && _MSC_FULL_VER >= 180030501) /* VS 2013 Update 2 */ \
    || XXH_HAS_INCLUDE(<avx2intrin.h>) /* GCC/Clang internal header */

